Doctrine/UserManager.php

```diff
@@ -28,30 +28,41 @@ class UserManager extends BaseUserManager
     /**
      * @var string
      */
+    private $className;
+
+    /**
+     * @var string
+     */
     protected $class;
 
     /**
      * @var ObjectRepository
      */
-    protected $repository;
+    private $repository;
 
     /**
      * Constructor.
      *
      * @param PasswordUpdaterInterface $passwordUpdater
      * @param CanonicalFieldsUpdater   $canonicalFieldsUpdater
      * @param ObjectManager            $om
-     * @param string                   $class
+     * @param string                   $className
      */
-    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $class)
+    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $className)
     {
         parent::__construct($passwordUpdater, $canonicalFieldsUpdater);
 
         $this->objectManager = $om;
-        $this->repository = $om->getRepository($class);
+        $this->className = $className;
```

Comment:
if we want to limit BC breaks, we could instead always resolve the class name in the constructor. This would require the loading of metadata only when using a Doctrine alias (which I want to deprecate)